### ISSUE TITLE
Enable kernel config required for DFS

### ISSUE THREAD

**Issue by jabbera** [microsoft/WSL#9524]
**Is your feature request related to a problem? Please describe.**
I'm unable to mount a CIFS DFS share because the kernel doesn't have support and DRVFS is too slow to be useful. This is being sold as the easy way to interop between linux and windows yet I can't mount one of the most basic enterprise shares.

**Describe the solution you'd like**
I'd the kernel to be DFS enabled.

**Describe alternatives you've considered**
DrvFS - too slow
Compiling my own kernel - Ew
Mounting the servers directly without DFS - also Ew

**Additional context**
Please enable the following kernel configs:

CONFIG_CIFS_UPCALL=y

**Comment by jabbera** [comment]
I'm not alone! https://github.com/microsoft/WSL/issues/5534

**Comment by elsaco** [comment]
IMO, your best option is `Ew`, compile your own kernel. Adding all sorts of storage protocols ( LIO was requested in https://github.com/microsoft/WSL/issues/9511 few days ago) makes the kernel blob larger. Soon we'll need 8GB of RAM to load the kernel only 🤷‍♂️

**Comment by jabbera** [comment]
I'm doing that now but considering dfs is a first class MSFT product offering I'd assume it would be supported in wsl

**Comment by jabbera** [comment]
This is blocked by: https://github.com/microsoft/WSL/issues/9540

**Comment by juliusl** [comment]
@elsaco For virtualization development 8GB of start-up RAM is pretty reasonable. For example, that's the amount needed to start windows server on Hyper-V. 

Would it be feasible to have multiple kernel builds then? 1 for default and another for "extended-io-features" or even "full-upstream-parity"?

**Comment by fpf3** [comment]
@jabbera 

Is this still blocked? Looks like #9540 got closed, but I don't see a PR.

**Comment by jabbera** [comment]
@fpf3 yes. This is blocked by https://github.com/microsoft/azurelinux/issues/8810 as well as the kernel config.
